include/fiction/io/print_layout.hpp

```diff
@@ -343,41 +353,147 @@ void print_charge_layout(std::ostream& os, const Lyt& lyt, const bool cs_color =
     static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
     static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
     static_assert(has_siqad_coord_v<Lyt>, "Lyt is not based on SiQAD coordinates");
-    static_assert(has_get_charge_state_v<Lyt>, "Lyt does not implement the get_charge_state function");
+    static_assert(has_get_charge_state_v<Lyt> | has_get_sidb_defect_v<Lyt>,
```

Comment:
While we're at this function, I would propose some slight rework here. This function should print any SiDB layout and only locally check if there is a charge interface or a defect interface and then print additional information respectively. It would make this function much cleaner, more concise, and more usable.

Reply:
so we delete line 356, since we already have ``static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout")`` where we check if we have a SiDB layout, right?